azimuth/modules/dataset_analysis/dataset_warnings.py

```diff
@@ -193,135 +195,117 @@ def get_general_warnings(
-        cls_index = np.arange(train_mng.get_num_classes(labels_only=True))
-
-        # Get token length and other (see screen 3b)
-        train_df = pd.DataFrame(
-            {
-                DatasetColumn.word_count: train_ds[DatasetColumn.word_count],
-                "label": train_ds[self.config.columns.label],
-            }
-        )
-        eval_df = pd.DataFrame(
-            {
-                DatasetColumn.word_count: eval_ds[DatasetColumn.word_count],
-                "label": eval_ds[self.config.columns.label],
-            }
-        )
-
-        # overall stats
-        train_overall = np.mean(cast(List[int], train_ds[DatasetColumn.word_count])), np.std(
-            cast(List[int], train_ds[DatasetColumn.word_count])
-        )
-        eval_overall = np.mean(cast(List[int], eval_ds[DatasetColumn.word_count])), np.std(
-            cast(List[int], eval_ds[DatasetColumn.word_count])
-        )
-
-        # Get mean and std token lengths
-        train_desc = train_df.groupby("label").agg(["mean", "std"])
-        eval_desc = eval_df.groupby("label").agg(["mean", "std"])
-        # Compute divergence
-        divergence = (train_desc - eval_desc)[DatasetColumn.word_count].apply(np.abs)
-        divergence = fill_at_missing_indices(divergence, find_missing_rows(divergence, cls_index))
-        divergence_mean = divergence["mean"].to_numpy()
-        divergence_std = divergence["std"].to_numpy()
-
-        # Make alerts from values.
-        max_delta_mean_words = int(self.config.dataset_warnings.max_delta_mean_words)
-        max_delta_std_words = int(self.config.dataset_warnings.max_delta_std_words)
-        alert_norm_mean = divergence_mean > max_delta_mean_words
-        alert_norm_std = divergence_std > max_delta_std_words
-        alert_norm = np.any([alert_norm_mean, alert_norm_std], axis=0)
+        class_index = np.arange(dm.get_num_classes(labels_only=True))
+        value_per_agg_per_split: Dict[DatasetSplitName, Dict[Agg, float]] = defaultdict(dict)
+        value_per_label_per_agg_per_split: Dict[
+            DatasetSplitName, Dict[Agg, np.ndarray]
+        ] = defaultdict(dict)
+        hist_per_label_per_split = {}
+
+        for split in self.available_dataset_splits:
+            ds = self.get_dataset_split(split).remove_columns([])
+            df = ds.remove_columns(
+                [
+                    column
+                    for column in ds.column_names
+                    if column not in [DatasetColumn.word_count, self.config.columns.label]
+                ]
+            ).to_pandas()
+
+            value_per_agg_per_split[split][Agg.mean] = df["word_count"].mean()
+            value_per_agg_per_split[split][Agg.std] = df["word_count"].std()
+
+            # Get mean and std word count per label
+            stats_per_label = df.groupby("label").agg(["mean", "std"]).reindex(class_index)
+            for agg in [Agg.mean, Agg.std]:
```

Comment:
Here, you could loop on `Agg`, instead of repeating the values.
```suggestion
            stats_per_label = df.groupby("label").agg(list(Agg)).reindex(class_index)
            for agg in Agg:
```

Reply:
Right thanks! I wanted to verify how to loop on an `Enum`, but forgot to check.